compiler/rustc_typeck/src/check/method/suggest.rs

```diff
@@ -449,6 +482,77 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 let mut label_span_not_found = || {
                     if unsatisfied_predicates.is_empty() {
                         err.span_label(span, format!("{item_kind} not found in `{ty_str}`"));
+                        if let ty::Adt(ref adt, _) = rcvr_ty.kind() {
+                            let mut inherent_impls_candidate = self
+                                .tcx
+                                .inherent_impls(adt.did)
+                                .iter()
+                                .copied()
+                                .filter(|def_id| {
+                                    if let Some(assoc) =
+                                        self.associated_item(*def_id, item_name, Namespace::ValueNS)
+                                    {
+                                        // Check for both mode is the same so we avoid suggesting
+                                        // incorect associated item.
+                                        match (mode, assoc.fn_has_self_parameter) {
+                                            (Mode::MethodCall, true) => {
+                                                if let SelfSource::MethodCall(_) = source {
+                                                    // We check that the suggest type is actually
+                                                    // different from the received one
+                                                    // So we avoid suggestion method with Box<Self>
+                                                    // for instance
+                                                    self.tcx.at(span).type_of(*def_id) != actual
+                                                        && self.tcx.at(span).type_of(*def_id)
+                                                            != rcvr_ty
+                                                } else {
+                                                    false
+                                                }
+                                            }
+                                            (Mode::Path, false) => true,
+                                            _ => false,
+                                        }
+                                    } else {
+                                        false
+                                    }
+                                })
+                                .collect::<Vec<_>>();
+                            if inherent_impls_candidate.len() > 0 {
+                                inherent_impls_candidate.sort();
+                                inherent_impls_candidate.dedup();
+                                // number of type to shows at most.
+                                const LIMIT: usize = 3;
```

Comment:
We usually show 4 elements when it is a list, I think 3 when they are part of a sentence, like here.